### PR TITLE
TelematicPlugin: Memory leak

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
Fix memory leaks issues with several variables created with malloc or new.

## Related Issue
https://github.com/usdot-fhwa-OPS/V2X-Hub/issues/575

## How Has This Been Tested?
Local integration test

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/v2i-hub/TelematicBridgePlugin/src/TelematicBridgeMsgWorker.h [hunk 1]
```diff
@@ -238,7 +240,10 @@ namespace TelematicBridge
             }
             else
             {
-                json["payload"] = StringToJson(cJSON_Print(msg->payload));
+                // Render a cJSON entity to text for transfer/storage. Free the char* when finished.
+                char * payloadPtr = cJSON_Print(msg->payload);
```
Comment (paulbourelly999) [inline src/v2i-hub/TelematicBridgePlugin/src/TelematicBridgeMsgWorker.h:244]: Can we replace this with a smart pointer to remove the need for free?

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
Please address sonar code smells and update PR description. It seems to be a copy of a previous issue.

File: src/v2i-hub/TelematicBridgePlugin/src/TelematicBridgeMsgWorker.h (hunk 1 above)
Comment (dan-du-car) [reply src/v2i-hub/TelematicBridgePlugin/src/TelematicBridgeMsgWorker.h:244]: It only returns a char* Type. How do we put this in smart pointer?

**Review by paulbourelly999 — APPROVED** [review]

File: src/v2i-hub/TelematicBridgePlugin/src/TelematicBridgeMsgWorker.h [hunk 2]
```diff
@@ -238,7 +240,9 @@ namespace TelematicBridge
             }
             else
             {
-                json["payload"] = StringToJson(cJSON_Print(msg->payload));
+                // Render a cJSON entity to text for transfer/storage. Free the char* when finished.
+                auto payloadPtr = std::make_unique<char*>(cJSON_Print(msg->payload));
```
Comment (paulbourelly999) [inline src/v2i-hub/TelematicBridgePlugin/src/TelematicBridgeMsgWorker.h:244]: Apparently `std::make_unique` is sort of a wrapper for `new`  (https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique) and actually creates a new object from the passed in parameter. This is likely why we still see the memory leak with this in. Apparently this is the more appropriate way to create a unique pointer from the returned char*
```cpp
 std::unique_ptr<char, decltype(std::free) *> t_copy { cJSON_Print(msg->payload), std::free };
```
Could you try this and see if it works?

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
See review

**Review by paulbourelly999 — APPROVED** [review]
Looks good, what happened to the smart pointer stuff?

**Pull request merged**